packages/next-app/src/components/PoolMetadataFormModal.tsx

```diff
@@ -1,23 +1,39 @@
 "use client";
-
 import * as Dialog from "@radix-ui/react-dialog";
 import { Cross2Icon } from "@radix-ui/react-icons";
 import cn from "classnames";
-import { useState } from "react";
+import cuid from "cuid";
+import * as React from "react";
+import { HTMLProps, useContext, useState } from "react";
+import { useForm } from "react-hook-form";
 
+import {
+  PoolMetadataAttribute,
+  PoolMetadataContext,
+} from "../contexts/PoolMetadataContext";
 import { Button } from "./Button";
 
-function Input({ label, ...rest }: React.HTMLProps<HTMLInputElement>) {
-  return (
-    <div className="mb-4">
-      <label className="mb-2 block text-sm text-gray-400">{label}</label>
-      <input
-        {...rest}
-        className="block w-full rounded border bg-gray-50 py-2 px-3 leading-tight text-gray-700 shadow focus:outline-none"
-      />
-    </div>
-  );
-}
+type Inputs = {
+  name: string;
+  desc: string;
+  type: string;
+  value: unknown;
+};
```

Comment:
couldn't you import `PoolMetadataAttribute`?

Reply:
`PoolMetadataAttribute` have an id, but the form input does not have an id. But TBH, it makes sense to add a hidden input field.

Reply:
cool! just remember that you can always use the utility-types [Pick](https://www.livelo.com.br/ganhe-pontos-compre-pontue-promo) and [Omit](https://www.livelo.com.br/ganhe-pontos-compre-pontue-promo) to construct new types from another one and avoid code duplication.

Reply:
cool! just remember that you can always use the utility-types [Pick](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) and [Omit](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) to construct new types from another one and avoid code duplication.